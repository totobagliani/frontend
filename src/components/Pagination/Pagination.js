import React from 'react';
import { ArrowLeftOutlined, ArrowRightOutlined } from '@ant-design/icons';
import styles from './styles.module.scss';

export default function Pagination() {
  return (
    <div className={styles.pagination}>
      <button type="button" className={styles.pagination__button}>
        <ArrowLeftOutlined className={styles.pagination__icon} />
      </button>
      <ul id="pagination" className={styles.pagination__list}>
        <li className={styles.pagination__item}>1</li>
        <li className={styles.pagination__item}>2</li>
        <li className={styles.pagination__item}>3</li>
        <li className={`${styles.pagination__item} ${styles.active}`}>4</li>
        <li className={styles.pagination__item}>5</li>
        <li className={styles.pagination__item}>6</li>
        <li className={styles.pagination__item}>7</li>
        <li className={styles.pagination__item}>8</li>
      </ul>
      <button type="button" className={styles.pagination__button}>
        <ArrowRightOutlined className={styles.pagination__icon} />
      </button>
    </div>
  );
}
