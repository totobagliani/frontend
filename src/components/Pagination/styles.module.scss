@import '../../styles/settings.scss';

.pagination {
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  border-top: 1px solid $gray-normal;
  border-bottom: 1px solid $gray-normal;
  margin: 1em 0;
  padding: 1em 0;
}

.pagination__list {
  list-style: none;
  display: flex;
  column-gap: 1em;
  font-family: $orderPaginationFont;
}
.pagination__item {
  cursor: pointer;
}

.active {
  font-weight: bolder;
}

.pagination__button {
  background-color: #fff;
  cursor: pointer;
  border: none;
  border-radius: 50%;
  padding: 0.5em;
  &:hover {
    background-color: $gray-normal;
  }
}
