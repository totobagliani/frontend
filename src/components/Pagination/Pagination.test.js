/* eslint-disable comma-dangle */
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import userEvent from '@testing-library/user-event';
import Pagination from './Pagination';
import {
  setCurrentPage,
  previousPage,
  nextPage
} from '../../redux/actions/ui.action';

const mockData = require('../../services/mockproducts.json');

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {
  products: {
    products: mockData.products,
    productAdded: {
      productName: '',
      description: '',
      imageURL: '',
      isFavourite: false,
      price: 0,
      section: ''
    },
    searchResults: {
      itemsCount: 3,
      searchTerm: 'algo'
    }
  },
  ui: {
    currentPage: 5,
    currentFilter: '',
    currentOrder: '',
    resultsVisible: true
  }
};

const store = mockStore(initialState);

store.dispatch = jest.fn();

jest.mock('../../redux/actions/ui.action', () => ({
  setCurrentPage: jest.fn(),
  previousPage: jest.fn(),
  nextPage: jest.fn()
}));

describe('Given the pagination Component', () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Pagination />
        </BrowserRouter>
      </Provider>
    );
    jest.clearAllMocks();
  });

  test('The component is redered (take a snapshot)', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Pagination />
      </Provider>
    );
    expect(asFragment(<Pagination />)).toMatchSnapshot();
  });
  test('when click the pages dispatch current page', () => {
    const pageNumber2 = screen.getAllByText(/2/)[0];
    userEvent.click(pageNumber2);
    expect(setCurrentPage).toHaveBeenCalled();
  });
  test('when click the next button dispatch next page', () => {
    const nextButton = screen.getAllByRole('button')[1];
    userEvent.click(nextButton);
    expect(nextPage).toHaveBeenCalledWith(initialState.ui.currentPage);
  });
  test('when click the previous button dispatch previous page', () => {
    const pageNumber3 = screen.getAllByText(/3/)[0];
    const prevButton = screen.getAllByRole('button')[0];

    userEvent.click(pageNumber3);

    userEvent.click(prevButton);
    expect(previousPage).toHaveBeenCalled();
  });
});
