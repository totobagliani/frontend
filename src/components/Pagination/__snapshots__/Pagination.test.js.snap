// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the pagination Component The component is redered (take a snapshot) 1`] = `
<DocumentFragment>
  <div
    class="pagination"
  >
    <button
      class="pagination__button"
      type="button"
    >
      <span
        aria-label="arrow-left"
        class="anticon anticon-arrow-left pagination__button"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="arrow-left"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 000 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </span>
    </button>
    <ul
      class="pagination__list"
      id="pagination"
    >
      <li
        class="pagination__item"
      >
        1
      </li>
    </ul>
    <button
      class="pagination__button"
      type="button"
    >
      <span
        aria-label="arrow-right"
        class="anticon anticon-arrow-right pagination__button"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="arrow-right"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M869 487.8L491.2 159.9c-2.9-2.5-6.6-3.9-10.5-3.9h-88.5c-7.4 0-10.8 9.2-5.2 14l350.2 304H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h585.1L386.9 854c-5.6 4.9-2.2 14 5.2 14h91.5c1.9 0 3.8-.7 5.2-2L869 536.2a32.07 32.07 0 000-48.4z"
          />
        </svg>
      </span>
    </button>
  </div>
</DocumentFragment>
`;
